### DISCUSSION TITLE
Linkerd Helm Race Condition when subcharted?

### DISCUSSION THREAD

**Discussion by agniveshadhikari** [linkerd/linkerd2#5108] [Q&A]
I am subcharting linkerd chart, and from the same uber level chart, I am also deploying ingress-nginx. I have added the necessary values for proxy injection into the nginx controller pods.

When I deploy, sometimes I see nginx pods don't have the proxy injected into them. I think this is because the nginx pods were created before the admission controller was ready, and hence, this is actually a race condition. Please correct me if I am wrong.

Is there a good way to do this? (Having one chart that deploys my whole application) Or do I **have** to put linkerd as a separate chart?

_I know it might be more of a helm question, but I still wanted to ask you guys._

**Comment by grampelberg** [comment]
You can:

- Set the MWC to `fail`. This'll force injection but introduce some other race conditions if you've not added labels correctly everywhere.
- Use the proxy partial like we do for the control plane.
- Add a post-install hook to the helm chart that restarts the ingress-nginx rollout.

Personally, I'd take the third option =)
